renderdoc/api/replay/structured_data.h

```diff
@@ -567,92 +617,142 @@ inline SDObject *makeSDObject(const char *name, QVariant val)
 #endif
 
 DOCUMENT("Make a structured object out of a signed integer");
-inline SDObject *makeSDInt64(const char *name, int64_t val)
+inline SDObject *makeSDInt64(const char *name, int64_t val, const char *customString,
+                             const char *customTypeName)
```

Comment:
Setting custom strings and custom typenames are a rare use and not something that every caller should have to specify `NULL` from C++ or `None` from python for if they're calling these directly. Ideally we'd just specify `NULL` default parameters or multiple parameter overloads like you do with `makeSDObject` but unfortunately while swig technically compiles those correctly it makes things pretty ugly so I try to avoid it for python-facing interfaces.

I think these should instead be moved to helper functions in `SDObject` like `SetTypeName` and `SetCustomString`, which are then also useful after creation. You could use the pattern to have them return `this` for chaining:

```cpp
SDObject *SDObject::SetTypeName(const char *customTypeName)
{
  type.name = customTypeName;
  return this;
}
SDObject *SDObject::SetCustomString(const char *customString)
{
  data.str = customString;
  type.flags = SDTypeFlags::HasCustomString;
  return this;
}

SDObject *foo = makeSDInt64("myInt", 500)->SetTypeName("stringInteger")->SetCustomString("fivehundred");
```

Reply:
done